lldb/source/Symbol/SwiftASTContext.cpp

```diff
@@ -1448,25 +1448,45 @@ void ApplyWorkingDir(SmallString &clang_argument, StringRef cur_working_dir) {
   llvm::sys::path::append(clang_argument, cur_working_dir, rel_path);
   llvm::sys::path::remove_dots(clang_argument);
 }
+
+enum ClangArgCategory {
+  ClangArgCategoryNone = 0,
+  ClangArgCategoryUnique = 1 << 0,
+  ClangArgCategoryMacro = 1 << 1,
+};
+
+inline bool IsMultiArgClangFlag(const StringRef& arg) {
+  return arg == "-D" || arg == "-U" || arg == "-working-directory";
+}
+
+ClangArgCategory GetClangArgCategory(const StringRef& arg) {
+  if (arg.size() <= 2 || arg[0] != '-')
+    return ClangArgCategoryNone;
+
+  if (arg[1] == 'D' || arg[1] == 'U')
+    return static_cast<ClangArgCategory>(
+        ClangArgCategoryUnique | ClangArgCategoryMacro);
+
+  return ClangArgCategoryNone;
+}
 } // namespace
 
 void SwiftASTContext::AddExtraClangArgs(std::vector<std::string> ExtraArgs) {
+  swift::ClangImporterOptions &importer_options = GetClangImporterOptions();
+  llvm::DenseSet<StringRef> unique_flags;
+  for (auto& arg : importer_options.ExtraArgs)
+    unique_flags.insert(arg);
```

Comment:
Will this add `-D MACRO` as `-D` and `MACRO`? Or are we guaranteed at this point that we contracted all multi-word arguments into single-word ones?

Reply:
Yeah, we're guaranteeing that all args that should be unique are joined by this line